src/server/types.ts

```diff
@@ -105,20 +118,33 @@ export interface Renderer {
   render(ctx: RenderContext, render: RenderFn): void;
 }
 
-export interface MiddlewareModule {
+// deno-lint-ignore no-explicit-any
+export interface MiddlewareModule<TState = any> {
   handler(
     req: Request,
-    ctx: MiddlewareHandlerContext,
+    ctx: MiddlewareHandlerContext<TState>,
   ): Response | Promise<Response>;
 }
 
-export interface Middleware {
+export interface Middleware<TState = Record<string, unknown>> {
   handler(
     req: Request,
-    ctx: MiddlewareHandlerContext,
+    ctx: MiddlewareHandlerContext<TState>,
   ): Response | Promise<Response>;
 }
 
+export interface MiddlewareRoute extends Middleware {
+  /**
+   * path-to-regexp style url path
+   */
+  route: string;
+  /**
+   * URLPattern of the route
+   */
+  // deno-lint-ignore no-explicit-any
+  regPattern: any;
```

Comment:
```suggestion
  pattern: URLPattern;
```

Reply:
On the server, `URLPattern` is declared in globals as `any` and so cannot be used as a type. I could declare it as a class instead but, I'm not sure there is lots of advantage. Having the proper definition would better, but i'm not too sure how.